File: Old python codes/Fixtrax_folder_path_extractor.py
from tkinter import *                     
from tkinter import filedialog as fd     
import os     

def openFile():
    the_file = fd.askdirectory(title = "Select the parent folder from which you want to extract the paths of all the Fixtrax folders")
    return the_file


input_folder = openFile()

output_file_path = str(input_folder) + "/Fixtrax_paths.csv"
with open(output_file_path, "w") as f:

    #Comment one of the following out according to your use case:

    #For 1 Day's experiments
    for subdir in os.listdir(input_folder):
        subdir_path = os.path.join(input_folder, subdir) 
        if os.path.isdir(subdir_path):
            for folder in os.listdir(subdir_path):
                folder_path = os.path.join(subdir_path, folder)
                if os.path.isdir(folder_path):
                    if "Fixtrax" in folder:
                        line = folder_path + ", test, FF0000\n"
                        f.write(line)

    # #For Combined of multiple day's experiments
    # for dir in os.listdir(input_folder):
    #     dir_path = os.path.join(input_folder, dir)
    #     if os.path.isdir(dir_path):
    #         for subdir in os.listdir(dir_path):
    #             subdir_path = os.path.join(dir_path, subdir) 
    #             if os.path.isdir(subdir_path):
    #                 for folder in os.listdir(subdir_path):
    #                     folder_path = os.path.join(subdir_path, folder)
    #                     if os.path.isdir(folder_path):
    #                         if "Fixtrax" in folder:
    #                             line = folder_path + ", test, FF0000\n"
    #                             f.write(line)

print("A text file containing all the paths to the JAABA folders in the given directory has been made")

File: Fixtrax_folder_path_extractor.m
function Fixtrax_folder_path_extractor()
    % Fixtrax Folder Path Extractor MATLAB script

    % Select the parent folder
    input_folder = uigetdir('', 'Select the parent folder from which you want to extract the paths of all the Fixtrax folders');
    
    if input_folder == 0
        disp('No folder selected. Exiting...');
        return;
    end
    
    output_file_path = fullfile(input_folder, 'Fixtrax_paths.csv');
    fid = fopen(output_file_path, 'w');
    if fid == -1
        error('Cannot create output file.');
    end

    % Uncomment one of the following blocks according to your use case:

    % For 1 Day's experiments
    subdirs = dir(input_folder);
    subdirs = subdirs([subdirs.isdir]); % Filter out non-directories
    subdirs = subdirs(~ismember({subdirs.name}, {'.', '..'})); % Remove '.' and '..'
    
    for i = 1:length(subdirs)
        subdir_path = fullfile(input_folder, subdirs(i).name);
        folders = dir(subdir_path);
        folders = folders([folders.isdir]); % Filter out non-directories
        folders = folders(~ismember({folders.name}, {'.', '..'})); % Remove '.' and '..'
        
        for j = 1:length(folders)
            folder_path = fullfile(subdir_path, folders(j).name);
            if contains(folders(j).name, 'Fixtrax')
                line = sprintf('%s, test, FF0000\n', folder_path);
                fprintf(fid, '%s', line);
            end
        end
    end

    % % For Combined of multiple day's experiments
    % dirs = dir(input_folder);
    % dirs = dirs([dirs.isdir]); % Filter out non-directories
    % dirs = dirs(~ismember({dirs.name}, {'.', '..'})); % Remove '.' and '..'
    % 
    % for i = 1:length(dirs)
    %     dir_path = fullfile(input_folder, dirs(i).name);
    %     subdirs = dir(dir_path);
    %     subdirs = subdirs([subdirs.isdir]); % Filter out non-directories
    %     subdirs = subdirs(~ismember({subdirs.name}, {'.', '..'})); % Remove '.' and '..'
    %     
    %     for j = 1:length(subdirs)
    %         subdir_path = fullfile(dir_path, subdirs(j).name);
    %         folders = dir(subdir_path);
    %         folders = folders([folders.isdir]); % Filter out non-directories
    %         folders = folders(~ismember({folders.name}, {'.', '..'})); % Remove '.' and '..'
    %         
    %         for k = 1:length(folders)
    %             folder_path = fullfile(subdir_path, folders(k).name);
    %             if contains(folders(k).name, 'Fixtrax')
    %                 line = sprintf('%s, test, FF0000\n', folder_path);
    %                 fprintf(fid, '%s', line);
    %             end
    %         end
    %     end
    % end
    
    fclose(fid);
    disp('A text file containing all the paths to the Fixtrax folders in the given directory has been made');
end
